### ISSUE TITLE
How can I iterate over an XMLRPCNode contents? [Not a bug]

### ISSUE THREAD

**Issue by SHxKM** [kodlian/AlamofireXMLRPC#6]
HI,

This is not a bug, I'm pretty new to Swift and AlamofireXMLRPC so I'm sure there's something wrong that I'm doing.

I have the following AFXMLRPC request:

```
AlamofireXMLRPC.request("\(WP_URL)", methodName: "wp.getTerms", parameters: params).responseXMLRPC { (response:Response<XMLRPCNode, NSError>) -> Void in
        guard response.result.isSuccess else {
            print(response.result.error)
            return
        }

        let davalue = response.result.value![0][0]["name"].string!

        print(davalue)
    } // End Alamofire request
```

The above code block works perfectly, and by that I mean that it gets me the first category in my site. As expected, assigning `devalue` the following:

`response.result.value![0][1]["name"].string!`

Gives me the 2nd category.

What I actually want is to iterate through my list of categories.

However, writing this:

```
let davalue = response.result.value![0]

for item in davalue {
     ….
}
```

Gives an error "Type 'XMLRPCNode" does not conform to protocol "SequenceType""

Is there any non-convoluted way to iterate through the values I'm getting **inside** the XMLRPCNode?

Thanks!

**Comment by kodlian** [comment]
You can't iterate directly on the node. However if the node is an array, you can access the swift collection directly:

``` swift
guard let param = response.result.value where param.count == 1 else {
   return
}
guard let collection = param[0].array else {
    return
}
for item in collection {
     ...           
}
```

**Comment by SHxKM** [comment]
Sorry, I've forgot another level of arrays in my original question.

I'm assuming `response.result.value` is an array of structs. (Return value of [wp.getTerms](https://codex.wordpress.org/XML-RPC_WordPress_API/Taxonomies#wp.getTerms))

When I give the request the following:

`let davalue = response.result.value![0][0]["name"].string!`

I get my first category as a string. I'm assuming it's accessing the node, then the first (and only?) array, then the struct.

However implementing your code I'm getting only my first category repeatedly.

Just so I make sure I'm actually learning something here:

When we write:

`param[0].array`

what does `.array` signify? what does it do?

I'm quite lost to be honest!

**Comment by SHxKM** [comment]
I got to this stage:

```
guard let param = response.result.value where param.count == 1 else {
            return
        }
        guard let collection = param[0].array else {
            return
        }
        var index = 0
        for item in collection {

            print(item["name"].string)
            index = index + 1

        }

        print(index)
```

`index` is for debugging and is showing 49, which is curiously the exact number of categories I have. I am baffled as to why `print(item["name"].string)` is printing the same (first) category name over and over again since it's clearly iterating over all categories!

**Comment by kodlian** [comment]
Did you get AlamofireXMLRPC from Cocoapods global repo or directly with the `:git => 'https://github.com/kodlian/AlamofireXMLRPC.git'`?

A fix has been merged this week but I have not pushed a new version yet to the global repo.

**Comment by SHxKM** [comment]
Thanks for sticking with me here!

This is what I have in my Podfile:

```
target 'App' do
    pod 'AlamofireXMLRPC', :git => 'https://github.com/kodlian/AlamofireXMLRPC.git'

    pod 'Alamofire', '~> 3.0'

end
```

**Comment by SHxKM** [comment]
> A fix has been merged this week but I have not pushed a new version yet to the global repo.

So is my Podfile config above correct? I've been banging my head with this so I'm wondering whether this could be a bug, after all? (even though the issue now isn't what I described in my OP) Why would it iterate over the exact number of categories, yet return the `["name"]` value of only the first one?

Update: This is happening to other methods supported by XMLRPC in WordPress, such as wp.getTerms → post-tag. Writing:

```
guard let param = response.result.value where param.count == 1 else {
            return
        }

        guard let collection = param[0].array else {
            return
        }


        for item in collection {
            self.catList.append(item["name"].string!)
        }
```

yields a collection with identical items, even though it correctly iterates X amount of times, indentical to the number of different items in the collection.

**Comment by SHxKM** [comment]
Sorry to bug you, but any updates? Is this bug or am I doing something wrong?

**Comment by SHxKM** [comment]
Ran the following to make sure I'm getting the last version:

```
pod repo remove master
pod setup
pod install
```

No dice.

**Comment by kodlian** [comment]
Could you provide an exemple of answer? I'm not able to help you right now but within a few days I will have some spare time and I will try to figure out what is going wrong.

**Comment by SHxKM** [comment]
I'm not sure what you mean by and example of answer, sorry I'm still quite a newbie.

Basically, here's the relevant part of my code:

```
AlamofireXMLRPC.request("\(WP_URL)", methodName: "wp.getTerms", parameters: params).responseXMLRPC {

        (response:Response<XMLRPCNode, NSError>) -> Void in
        guard response.result.isSuccess else {
            print(response.result.error)
            return
        }


        guard let param = response.result.value where param.count == 1 else {
            return
        }

        guard let collection = param[0].array else {
            return
        }


        for item in collection {
            self.tagList.append(item["name"].string!)

        }
```

Of, both `tagList` and `catList` are arrays of Strings.

When I print `self.tagList.count`, I get the correct count of tags (in my case, 367). That's also true if I try to print `self.catList.count` in the method I showed above (in my case, there are 49 categories). But when I print the items themselves that I'm appending to `tagList` / `catList`, I'm getting 367 or 49 times the same item, and that is always the first item (at index `0`) of the array.

So my first category is `Apps`, so printing `catList` gives me an array of strings that are all `Apps`. My first tag is `"example"`, and when printing each item in  `tagList` I'm getting an array of strings that are all `"example"`. 

At first I thought I'm going one level too deep, but printing the count — which is the same as the category and tag count in my site (I checked) — made me certain that it is indeed going through the **correct number** of items, it's just always appending the same item, the first item.

It seems to me that this is the same problem that is documented in issue #3 .

**Comment by kodlian** [comment]
I have clearly understand your issue and this is really strange since the fix was just made for that. For the answer, I just need an exemple of the XMLRPC received from the server. This will allow me to try myself to parse the XML RPC answer.

**Comment by SHxKM** [comment]
> I have clearly understand your issue and this is really strange since the fix was just made for that. For the answer, I just need an exemple of the XMLRPC received from the server. This will allow me to try myself to parse the XML RPC answer.

Ok, can you guide me through on how to supply such thing? is it supposed to be a file or a screenshot from my debug window? I'm assuming it's supposed to be a file so I'm not sure how to do that in Xcode and Swift. I will supply this file immediately so you can have a look when you're free. Thanks.

**Comment by kodlian** [comment]
Instead of using responseXMLRPC, use the Alamofire's responseString method. And you will get the string representation of the server answer.

**Comment by SHxKM** [comment]
I hope this is good enough, let me know if you need the whole thing:

[example.txt](https://github.com/kodlian/AlamofireXMLRPC/files/71321/example.txt)

**Comment by kodlian** [comment]
It's ok. Thanks.

**Comment by SHxKM** [comment]
**Update**: I've managed to solve it, there was something wrong with my pod install! I'm very sorry, but I ran `pod update AlamofireXMLRPC` and it says "nothing to update", then I typed `pod outdated` and it updated Alamofire itself (from 3.1.3 to 3.1.4), then it said it also updated AlamofireXMLRPC from (1.0.0 to 1.0.0. Here's my Terminal output:

pod outdated
Updating spec repo `master`
Analyzing dependencies
The following pod updates are available:
- Alamofire 3.1.3 -> 3.1.4 (latest version 3.1.4)

pod update AlamofireXMLRPC
Updating local specs repositories
Analyzing dependencies
Pre-downloading: `AlamofireXMLRPC` from `https://github.com/kodlian/AlamofireXMLRPC.git`
Downloading dependencies
Using AEXML (2.0.1)
Installing Alamofire 3.1.4 (was 3.1.3)
Installing AlamofireXMLRPC 1.0.0 (was 1.0.0)
Generating Pods project
Integrating client project
Sending stats
Pod installation complete! There are 2 dependencies from the Podfile and 3 total
pods installed.

I thought `pod update` wasn't needed if I have the URL in the Podfile…?

Anyway, sorry for wasting your time and thank you **very much** for this plugin and your time here!
